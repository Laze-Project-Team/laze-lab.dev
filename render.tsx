import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
