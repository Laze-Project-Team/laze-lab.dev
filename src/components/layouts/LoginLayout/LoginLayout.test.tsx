import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { render } from '/render';

import { LoginLayout } from '.';

it('should be rendered correctly', () => {
  const { container } = render(<LoginLayout title="title" />);
  expect(container.firstChild).toMatchSnapshot();
});

it('should have "main" role', () => {
  render(<LoginLayout title="title" />);
  expect(screen.getByRole('main')).toBeInTheDocument();
});

it('should render children', () => {
  const children = "Hey I'm children";
  render(<LoginLayout title="title">{children}</LoginLayout>);
  expect(screen.getByText(children)).toBeInTheDocument();
});
