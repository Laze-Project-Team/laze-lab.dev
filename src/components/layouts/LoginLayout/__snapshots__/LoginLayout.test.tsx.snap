// Vitest Snapshot v1

exports[`should be rendered correctly 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
>
  <main>
    <h1
      class="css-1pqndvo-PresentialLoginLayout"
    >
      title
    </h1>
    <a
      class="css-1ln0tcx-PresentialStyledLink-PresentialStyledLink-PresentialDefaultLink-PresentialDefaultLink"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17e890r-MuiSvgIcon-root"
        data-testid="ArrowBackIosIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
        />
      </svg>
      <span>
        backToHome
      </span>
    </a>
  </main>
</div>
`;
